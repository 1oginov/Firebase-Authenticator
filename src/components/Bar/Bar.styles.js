// @flow

import { withStyles, type Theme } from '@material-ui/core/styles';

const styles = (theme: Theme) => ({
  actionItems: {
    flexShrink: 0,
    marginRight: -1.5 * theme.spacing.unit,
  },
  navigationButton: {
    marginLeft: -1.5 * theme.spacing.unit,
    marginRight: 2.5 * theme.spacing.unit,
  },
  title: {
    flexGrow: 1,
  },
});

export default withStyles(styles);
