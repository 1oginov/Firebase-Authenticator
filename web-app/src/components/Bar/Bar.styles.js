// @flow

import { withStyles, type Theme } from '@material-ui/core/styles';

const styles = (theme: Theme) => ({
  actionItems: {
    flexShrink: 0,
    marginRight: theme.spacing(-1.5),
  },
  navigationButton: {
    marginLeft: theme.spacing(-1.5),
    marginRight: theme.spacing(2.5),
  },
  title: {
    flexGrow: 1,
  },
});

export default withStyles(styles);
