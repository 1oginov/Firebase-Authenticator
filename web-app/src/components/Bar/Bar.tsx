import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import * as React from 'react';

type Props = {
  actionItems?: ?Array<{
    click: () => void,
    icon: React.Node,
  }>,
  classes: { [string]: string },
  navigationClick: () => void,
  navigationIcon?: ?React.Node,
  title: string,
};

const Bar = ({
  actionItems, classes, navigationClick, navigationIcon, title,
}: Props) => (
  <AppBar color="default" position="static">
    <Toolbar>

      <IconButton className={classes.navigationButton} color="inherit" onClick={navigationClick}>
        {navigationIcon || <ArrowBackIcon />}
      </IconButton>

      <Typography className={classes.title} color="inherit" noWrap variant="h6">
        {title}
      </Typography>

      {actionItems && actionItems.length > 0 && (
        <div className={classes.actionItems}>
          {actionItems.map(({ click, icon }, index) => (
            <IconButton
              color="inherit"
              key={index} // eslint-disable-line react/no-array-index-key
              onClick={click}
            >
              {icon}
            </IconButton>
          ))}
        </div>
      )}

    </Toolbar>
  </AppBar>
);

Bar.defaultProps = {
  actionItems: undefined,
  navigationIcon: undefined,
};

export default Bar;
