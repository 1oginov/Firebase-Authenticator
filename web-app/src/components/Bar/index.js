// @flow

import Bar from './Bar';
import style from './Bar.styles';

export default style(Bar);
