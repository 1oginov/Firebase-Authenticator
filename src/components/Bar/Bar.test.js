// @flow

import classes from 'identity-obj-proxy';
import { shallow } from 'enzyme';
import React from 'react';

import Bar from './Bar';

describe('<Bar />', () => {
  it('renders without crashing and matches snapshot', () => {
    const wrapper = shallow(
      <Bar
        classes={classes}
        navigationClick={() => undefined}
        title="Title"
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
