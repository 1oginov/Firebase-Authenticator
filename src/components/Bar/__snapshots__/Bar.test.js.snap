// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar /> renders without crashing and matches snapshot 1`] = `
<WithStyles(AppBar)
  color="default"
  position="static"
>
  <WithStyles(Toolbar)>
    <WithStyles(IconButton)
      className="navigationButton"
      color="inherit"
      onClick={[Function]}
    >
      <pure(ArrowBackIcon) />
    </WithStyles(IconButton)>
    <WithStyles(Typography)
      className="title"
      color="inherit"
      noWrap={true}
      variant="h6"
    >
      Title
    </WithStyles(Typography)>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
